coalib/bearlib/aspects/Redundancy.py

```diff
@@ -0,0 +1,255 @@
+        Subversion.
+        """
+    )
+)
+
+Root.Redundancy.UnreachableCode.new_subaspect(
+    'UnreachableStatement',
+    AspectDocumentation(
+        definition="""
+        An unreachable statement is a statement that is never executed
+        during code execution.
+        """,
+        example="""
+        def func():
+            return True
+
+        if func():
+            a = {}
+        else:
+            a = (i for i in range (5))
+            print (id(a))
+        """,
+        example_language="python",
+        fix_suggestions="""
+        These statement can be remove without harming the code.
+        """
+    )
+)
+
+Root.Redundancy.new_subaspect(
+    'UnusedVariable',
+    AspectDocumentation(
+        definition="""
+        Unused variables are declared but never used.
+        """,
+        example="""
+        a = {}
+        print ('coala')
+        """,
+        example_language='python',
+        importance_reason="""
+        Unused variables can degrade performance marginally but more importantly
+        makes the source code harder to read and understand.
+        """,
+        fix_suggestions="""
+        Those variables can easily be removed without consequences.
+        """
+    )
+)
+
+Root.Redundancy.UnusedVariable.new_subaspect(
+    'UnusedParameter',
+    AspectDocumentation(
+        definition="""
+        Unused parameters are functions arguments which are never used.
+        """,
+        example="""
+        def func(a):
+            pass
+        """,
+        example_language='python',
+        importance_reason="""
+        Unused paramaters are useless to functions, they them difficult to
+        use and maintain.
+        """,
+        fix_suggestions="""
+        Those parameters can easily be removed without consequences.
+        """
+    )
+)
+
+
+Root.Redundancy.UnusedVariable.new_subaspect(
+    'UnusedLocalVariable',
+    AspectDocumentation(
+        definition="""
+        These are variable which are defined locally but never used.
+        """,
+        example="""
+        for i in range (5):
+            a = 0
+            print ( ' coala ' )
+        """,
+        example_language='python',
+        importance_reason="""
+        They make the code difficult to maintain.
+        """,
+        fix_suggestions="""
+        These can easily be removed without consequences.
+        """
+    )
+)
+Root.Redundancy.UnusedVariable.new_subaspect(
+    'UnusedGlobalVariable',
+    AspectDocumentation(
+        definition="""
+        These are variable which have a global scope but are never used.
+        """,
+        example="""
+        a = 0
+        for i in range (5):
+            print ( ' coala ' )
+        """,
+        example_language='python',
+        importance_reason="""
+        They make the code difficult to maintain.
+        """,
+        fix_suggestions="""
+        These can easily be removed without consequences.
+        """
+    )
+)
```

Comment:
I think we might want to rethink the UnusedGlobalVariable vs. UnusedLocalVariable one... there are many other possible scopes, is this really the way to go? Maybe remove them for now and see what we need to add when we need it for a bear?